site/docs/extensions/generate_function_docs.py

```diff
@@ -37,6 +37,8 @@ def write_markdown(file_obj: dict, file_name: str) -> None:
             options_list = []
 
             for count, impls in enumerate(implementations_list):
+                if "args" not in impls:
```

Comment:
AFAICT, this is not reading the "options" field of `impls` so we won't get any doc from that. Furthermore, I would expect `options_list` to be constructed from each `impls["options"]` entry since each impl may define its own options. Instead, since that variable is outside the loop it is accumulating the union of all options from anything in `implementation_list`- we haven't noticed because most impls of a single function have identical options.

It might be desirous to *require* that all impls have identical options, possibly by hoisting options to a property of the function rather than of the impl. That adds some validation overhead when multiple function declarations share a name- what should be the behavior if regular "add" and decimal "add" declare different options? I'd say keeping options as a property of `impls[n]` is probably preferable

Reply:
I'll fix this up so we generate docs for options.  Yes, we will want options to be per-impl I think.  There is some precedent for this already.  Integer arithmetic usually has overflow for example while fp arithmetic had rounding mode.

Reply:
Ok.  I restored it to what it used to be.  However, I left it per-impl.  This can go wrong as it does here:

![image](https://user-images.githubusercontent.com/1696093/196623089-21e38045-6b31-4fbd-8050-e47707db5a41.png)

However, that can be left for a future issue.